### PR TITLE
[BugFix] Clear rowset del_vec cache of primary key table after alter job finish

### PR DESCRIPTION
**Pull request by sevev**
When a primary key tablet is doing alter job, we will convert historical rowsets, clear historical tablet meta, rebuild tablet meta, and reload it. However, we do not clear the del_vec cache of the tablet. So we may get the expired del_vec from del_vec cache and may cause read data error.

## What type of PR is this:
- [x] BugFix
- [ ] UT
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -251,6 +251,13 @@ Status TabletUpdates::_load_from_pb(const TabletUpdatesPB& tablet_updates_pb) {
     std::unique_lock l1(_lock);
     std::unique_lock l2(_rowsets_lock);
 
+    std::vector<TabletSegmentId> tsids;
```
Comment (chaoyli) [inline be/src/storage/tablet_updates.cpp:254]: What's the error raised?
It's like this one?
```
java.sql.SQLException: (conn=448) Bad page: invalid footer, file=/opt/starrocks/be/storage/data/812/2360503/1912606354/020000000003b5eeff491c6513959632d141c7e3baca8c9d_0.dat
```

Comment (sevev) [reply be/src/storage/tablet_updates.cpp:254]: > What's the error raised? It's like this one?
> 
> ```
> java.sql.SQLException: (conn=448) Bad page: invalid footer, file=/opt/starrocks/be/storage/data/812/2360503/1912606354/020000000003b5eeff491c6513959632d141c7e3baca8c9d_0.dat
> ```

No, the error maybe like `actual row size changed` or return error query result. It does not cause disk file error but maybe lost some data.

**Review by chaoyli — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**